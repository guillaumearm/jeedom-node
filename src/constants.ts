export const JEEDOM_API_ENDPOINT = '/core/api/jeeApi.php'
