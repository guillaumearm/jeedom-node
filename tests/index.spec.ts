import { setupRecorder } from 'nock-record'
import { identity, evolve, take, pipe, map, assoc } from 'ramda'
import { createRecordTest } from './utils'
import Jeedom from '../src'
import { JEEDOM_HOST, JEEDOM_API_KEY } from './constants'

type GenericTransform = (x: any) => any
const prepareGenericData = (transform: GenericTransform) => pipe(
  take(1),
  map(pipe(
    transform,
    assoc('display', {}),
    assoc('configuration', {}),
  )),
)

describe('jeedom-node', () => {
  describe('exports', () => {
    test('default export should be a function', () => {
      expect(typeof Jeedom).toBe('function')
    })
  })

  describe('api', () => {
    const recordTest = createRecordTest('jeedom', setupRecorder({ mode: 'record' }))

    const api = Jeedom({ host: JEEDOM_HOST, apikey: JEEDOM_API_KEY })

    recordTest('ping', async () => {
      expect(await api.ping()).toMatchSnapshot()
    })
    recordTest('version', async () => {
      expect(await api.version()).toMatchSnapshot()
    })
    recordTest('datetime', async () => {
      expect(await api.datetime()).toMatchSnapshot()
    })

    describe('config', () => {
      recordTest('config::byKey', async () => {
        expect(await api.config.byKey({ key: 'name' })).toMatchSnapshot()
      })

      recordTest('config::save', async () => {
        const res1 = await api.config.save({ plugin: '__test__', key: '__test__', value: 'test' })
        const value = await api.config.byKey({ plugin: '__test__', key: '__test__' })
        const res2 = await api.config.save({ plugin: '__test__', key: '__test__', value: '' })
        expect(value).toBe('test')
        expect([res1, value, res2]).toMatchSnapshot()
      })
    })

    describe('event', () => {
      recordTest('event::changes', async () => {
        const changes = await api.event.changes()
        const prepareData = evolve({ result: take(1) })

        expect(prepareData(changes)).toMatchSnapshot()
      })
    })

    describe('object', () => {
      recordTest('object::all', async () => {
        const objects = await api.object.all()
        const prepareData = prepareGenericData(identity)

        expect(prepareData(objects)).toMatchSnapshot()
      })

      recordTest('object::full', async () => {
        const fullObjects = await api.object.full()

        const prepareData = prepareGenericData(evolve({
          eqLogics: prepareGenericData(evolve({
            cmds: prepareGenericData(identity),
          })),
        }))

        expect(prepareData(fullObjects)).toMatchSnapshot()
      })
    })
  })
})
