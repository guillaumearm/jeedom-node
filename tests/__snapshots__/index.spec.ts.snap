// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jeedom-node api config config::byKey 1`] = `"Jeedom"`;

exports[`jeedom-node api config config::save 1`] = `
Array [
  null,
  "test",
  null,
]
`;

exports[`jeedom-node api datetime 1`] = `1541326095.9178`;

exports[`jeedom-node api event event::changes 1`] = `
Object {
  "datetime": 1541326094.048,
  "result": Array [
    Object {
      "datetime": 1541325903.4416,
      "name": "eqLogic::update",
      "option": Object {
        "eqLogic_id": "97",
      },
    },
  ],
}
`;

exports[`jeedom-node api object object::all 1`] = `
Array [
  Object {
    "configuration": Object {},
    "display": Object {},
    "father_id": null,
    "id": "13",
    "isVisible": "1",
    "name": "Automatismes",
    "position": null,
  },
]
`;

exports[`jeedom-node api ping 1`] = `"pong"`;

exports[`jeedom-node api version 1`] = `"3.2.11"`;
